src/main/webapp/app/exercises/programming/manage/update/update-components/information/programming-exercise-information.component.ts

```diff
@@ -7,43 +7,90 @@ import { Subject, Subscription } from 'rxjs';
 import { TableEditableFieldComponent } from 'app/shared/table/table-editable-field.component';
 import { every } from 'lodash-es';
 import { ImportOptions } from 'app/types/programming-exercises';
+import { ProgrammingExerciseInputField } from 'app/exercises/programming/manage/update/programming-exercise-update.helper';
+import { removeSpecialCharacters } from 'app/shared/util/utils';
 
 @Component({
     selector: 'jhi-programming-exercise-info',
     templateUrl: './programming-exercise-information.component.html',
-    styleUrls: ['../../programming-exercise-form.scss', 'programming-exercise-information.component.scss'],
+    styleUrls: ['../../../programming-exercise-form.scss', 'programming-exercise-information.component.scss'],
 })
-export class ProgrammingExerciseInformationComponent implements AfterViewInit, OnDestroy {
+export class ProgrammingExerciseInformationComponent implements AfterViewInit, OnChanges, OnDestroy {
+    protected readonly ProjectType = ProjectType;
+
     @Input() isImport: boolean;
     @Input() isExamMode: boolean;
     @Input() programmingExercise: ProgrammingExercise;
     @Input() programmingExerciseCreationConfig: ProgrammingExerciseCreationConfig;
     @Input() isLocal: boolean;
     @Input() importOptions: ImportOptions;
+    isSimpleMode = input.required<boolean>();
```

Comment:
is it possible to change the other inputs also to signals? Or is implies that several other changes?

Reply:
I have looked at it and converted all variables except for `programmingExerciseCreationConfig` to input signals.

Due to the poor approach of calculating `programmingExerciseCreationConfig` in the parent component we cannot convert it to a signal here before making sure to convert the creationConfig to a signal aswell, which I see as out of scope for this PR.